scripts/LSTM.py

```diff
@@ -1,315 +1,289 @@
+        remove_obvious=remove_obvious,
+        clean=clean,
+    )
+    filepath = (
+        "../data/"
+        + week_type
+        + "-"
+        + feature_types[0]
+        + "-"
+        + course
+        + "/feature_labels.csv"
+    )
+    labels[course] = load_labels(filepath, cem=cem, normalize=normalize)
+    feats, sel_feats, num_weeks, num_features = load_features(
+        filepath,
+        feature_types,
+        week_type,
+        course,
+        feature_names,
+        remove_obvious=remove_obvious,
+        fill=fill,
+        drop=drop,
+        normalize=normalize,
+    )
+    features[course] = feats
+    selected_features[course] = sel_feats
+    file = "selected_features/" + course + ".json"
+    with open(file, "w") as f:
+        json.dump(selected_features[course], f)
+    params[course]["num_weeks"] = num_weeks
+    params[course]["num_features"] = num_features
 
 # training models
 
 for course in courses:
-
-  fts = features[course].copy()
-  target = labels[course]
-  fts = fts.reshape(fts.shape[0], -1)
-  train_size = 0.8
-  x_train, x_rem, y_train, y_rem = train_test_split(fts, target, train_size=train_size, random_state=25)
-  x_test, x_val, y_test, y_val = train_test_split(x_rem, y_rem, train_size=0.5, random_state=25)
-  print(course+':')
-  print(x_train.shape,x_test.shape,x_val.shape)
-  print(y_train.shape,y_test.shape,y_val.shape)
-
-
-  num_weeks = params[course]['num_weeks']
-  num_features = params[course]['num_features']
-  num_epochs = params[course]['num_epochs']
-
-  current_timestamp = str(time.time())[:-2]
-  model=bidirectional_lstm
-  print(model.__name__)
-  history, scores = model(x_train, y_train, x_test, y_test, x_val, y_val,
-                          week_type, feature_types, course, n_weeks=num_weeks,
-                          n_features=num_features, num_epochs=num_epochs)
-  print("{:<15} {:<8} ".format('metric','value'))
-  for ke, v in scores.items():
-    if isinstance(v, float):
-        v=round(v, 4)
-    if ke!="feature_type":
-        print("{:<15} {:<8} ".format(ke, v))
-  run_name = model.__name__ + "_" + course + "_" + current_timestamp
-  fig = plot_history(history=history, filename=run_name)
+    fts = features[course].copy()
+    target = labels[course]
+    fts = fts.reshape(fts.shape[0], -1)
+    train_size = 0.8
+    x_train, x_rem, y_train, y_rem = train_test_split(
+        fts, target, train_size=train_size, random_state=25
+    )
+    x_test, x_val, y_test, y_val = train_test_split(
+        x_rem, y_rem, train_size=0.5, random_state=25
+    )
+    print(course + ":")
+    print(x_train.shape, x_test.shape, x_val.shape)
+    print(y_train.shape, y_test.shape, y_val.shape)
+
+    num_weeks = params[course]["num_weeks"]
+    num_features = params[course]["num_features"]
+    num_epochs = params[course]["num_epochs"]
+
+    current_timestamp = str(time.time())[:-2]
+    model = bidirectional_lstm
+    print(model.__name__)
+    history, scores = model(
```

Comment:
Likewise here, I would divide it into smaller chunks so the model call line is not so long -- i.e. x_train, y_train, (newline) x_test, y_test (newline) etc.